Fix bug simplifying if-then-else when it can be removed

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44462 Fix bug simplifying if-then-else when it can be removed**

Differential Revision: [D23671157](https://our.internmc.facebook.com/intern/diff/D23671157)

bertmaher committed:
Fix bug simplifying if-then-else when it can be removed

[ghstack-poisoned]

bertmaher committed:
Update on "Fix bug simplifying if-then-else when it can be removed"




[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Could we also have a test for this case please?

bertmaher commented:
Yes, let me see if I can cook one up.  I unfortunately stumbled across this while doing something mildly complicated :-)

bertmaher committed:
Found another bug while writing a test.  Ever shake a haystack and have needles pop out at your feet? on "Fix bug simplifying if-then-else when it can be removed"


Differential Revision: [D23671157](https://our.internmc.facebook.com/intern/diff/D23671157)

[ghstack-poisoned]
